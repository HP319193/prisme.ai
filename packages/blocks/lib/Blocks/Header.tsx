import { useBlock } from '../Provider';
import { useBlocks } from '../Provider/blocksContext';

import { Action, ActionConfig, ActionProps } from './Action';
import { BaseBlockConfig } from './types';
import { BaseBlock } from './BaseBlock';
import { useCallback, useEffect, useMemo, useState } from 'react';
import useLocalizedText from '../useLocalizedText';
import { BlocksListConfig, BlocksList } from './BlocksList';
import { CloseOutlined, MenuOutlined } from '@ant-design/icons';

interface HeaderConfig extends BaseBlockConfig {
  title?: Prismeai.LocalizedText;
  level: 1 | 2 | 3 | 4 | 5 | 6;
  logo?: {
    src: string;
    alt: string;
    action?: Omit<ActionConfig, 'text'>;
  };
  nav: (ActionConfig | BlocksListConfig)[];
  fixed?: boolean;
}

interface HeaderProps
  extends HeaderConfig,
    Pick<ActionProps, 'Link' | 'events'> {}

const isAction = (
  item: ActionConfig | BlocksListConfig
): item is ActionConfig => {
  return !(item as BlocksListConfig).blocks;
};

interface MenuItemProps extends Pick<ActionProps, 'Link' | 'events'> {
  item: HeaderConfig['nav'][number];
}
const MenuItem = ({ item, events, Link }: MenuItemProps) => {
  if (isAction(item)) {
    return <Action {...item} events={events} Link={Link} />;
  }
  return <BlocksList {...item} />;
};

export const Header = ({
  Link,
  events,
  className,
  level = 1,
  fixed,
  ...config
}: HeaderProps) => {
  const { localize } = useLocalizedText();
  const [visible, setVisible] = useState(false);
  const [hasScrolled, setHasScrolled] = useState(false);
  const nav = config.nav && Array.isArray(config.nav) ? config.nav : [];

  useEffect(() => {
    const listener = () => {
      setHasScrolled(window.scrollY > 0);
    };
    window.addEventListener('scroll', listener);
    return () => {
      window.removeEventListener('scroll', listener);
    };
  });

  const logo = useMemo(
    () =>
      config.logo ? (
        <img
          src={config.logo.src}
          alt={config.logo.alt}
          className="pr-block-header__logo-image"
        />
      ) : null,
    [config.logo]
  );

  const H = `h${level}` as keyof JSX.IntrinsicElements;
  const title = localize(config.title);

  const toggle = useCallback(() => {
    setVisible((visible) => !visible);
  }, []);

  return (
    <div
      className={`pr-block-header ${className} ${
        fixed ? 'pr-block-header--fixed' : ''
      } ${hasScrolled ? 'pr-block-header--has-scrolled' : ''}`}
    >
      <div className="pr-block-header__container">
        <div className="pr-block-header__left">
          {logo && (
            <div className="pr-block-header__logo">
              {config.logo?.action && (
                <Action
                  {...config.logo.action}
                  text={logo}
                  Link={Link}
                  events={events}
                />
              )}
              {!config.logo?.action && logo}
            </div>
          )}
          {title && <H className="pr-block-header__title">{title}</H>}
        </div>
        <div className="pr-block-header__right">
          <button className="pr-block__menu-toggle" onClick={toggle}>
            <MenuOutlined />
          </button>
          <div
            className={`pr-block__menu ${
              visible ? 'pr-block__menu--visible' : 'pr-block__menu--hidden'
            }`}
          >
            {nav.map((item, k) => (
              <div key={k} className="pr-block__menu-item">
                <MenuItem item={item} events={events} Link={Link} />
              </div>
            ))}
            <button
              className="pr-block__menu-toggle pr-block__menu-toggle--close"
              onClick={toggle}
            >
              <CloseOutlined />
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

const defaultStyles = `:block {
  min-height: 4rem;
  --pr-header-color: var(--color-accent, rgba(0,0,0,.8));
  --pr-header-contrast: var(--color-accent-contrast, white);
  --pr-header-logo-height: 3rem;
}
.pr-block-header__container {
  display: flex;
  flex: 1 1 0%;
  justify-content: space-between;
  position: relative;
  background-color: var(--pr-header-contrast);
}
:block.pr-block-header--fixed .pr-block-header__container {
  position: fixed;
  left: 0;
  right: 0;
  transition: background-color .2s ease-in;
}
:block.pr-block-header--fixed.pr-block-header--has-scrolled .pr-block-header__container {
  background-color: var(--pr-header-color);
}
:block.pr-block-header--fixed.pr-block-header--has-scrolled .pr-block-header__container *:not(svg, path) {
  color: var(--pr-header-contrast);
}

@media (min-width:768px) {
  :block {
    align-items: center;
  }
}

.pr-block-header__left {
  display: flex;
  margin: 1rem;
  z-index: 1;
}

.pr-block-header__logo,
.pr-block-header__logo a {
  display: flex;
  justify-content: center;
}
.pr-block-header__logo {
  margin-right: 1rem;
  height: var(--pr-header-logo-height, 3rem);
}

.pr-block-header__logo-image {
  height: 100%;
}

.pr-block-header__title {
  display: flex;
  align-items: center;
  margin: 0;
  font-weight: 700;
}

.pr-block-header__right {
  display: flex;
  justify-content: flex-end;
  color: var(--color-text);
  background-color: transparent;
}
.pr-block__menu {
  display: flex;
  flex-direction: row;
}
.pr-block__menu-toggle {
  display: none;
}
.pr-block__menu-item {
  padding: 0.5rem 1rem;
  display: flex;
  align-items: center;
}
@media (max-width: 414px) {
  :block {
    height: 4rem;
  }
  .pr-block__menu {
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    background-color: var(--pr-header-color);
    transition: transform .2s ease-in;
    transform: translate3d(0, -100%, 0);
    padding-top: 6rem;
    height: 100vh;
    flex-direction: column;
    color: var(----pr-header-contrast);
    padding: 10rem 2rem 0 2rem;
  }
  .pr-block-header__container *:not(svg, path) {
    color: var(--pr-header-contrast);
  }
  .pr-block__menu--visible {
    transform: translate3d(0, 0, 0);
  }
  .pr-block__menu-toggle {
    display: flex;
    position: absolute;
    top: 1rem;
    right: 1rem;
    font-size: 1.8rem;
  }
}
`;

export const HeaderInContext = () => {
  const { config, events } = useBlock<HeaderConfig>();
  const {
    components: { Link },
  } = useBlocks();

  return (
    <BaseBlock defaultStyles={defaultStyles}>
      <Header {...config} Link={Link} events={events} />
    </BaseBlock>
  );
};

HeaderInContext.styles = defaultStyles;

export default HeaderInContext;
