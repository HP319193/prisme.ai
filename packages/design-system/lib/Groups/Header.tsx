import { Button, Space, Avatar, Divider, Dropdown, Menu } from '../index';
import { FC, useMemo } from 'react';
import { ShareAltOutlined } from '@ant-design/icons';
import { ReactElement } from 'react';

export interface HeaderProps {
  workspaces: string[];
  t: Function;
  userName: string;
  userAvatar: string;
  icon?: ReactElement;
}

const Header = ({ workspaces, t, userName, userAvatar, icon }: HeaderProps) => {
  const workspacesMenu = useMemo(
    () => <Menu items={workspaces} onClick={() => {}} />,
    [workspaces]
  );

  return (
    <div className="px-6 flex flex-row w-full justify-between items-center pr-header">
      {icon}
      <Dropdown Menu={workspacesMenu}>{workspaces[0]}</Dropdown>
      <div className="flex flex-row items-center">
        <Button variant="grey">
          <Space>
            {t('shareText')}
            <ShareAltOutlined />
          </Space>
        </Button>
        <Divider type="vertical" className="mr-4" />
        <Space>
          {userName}
          <Avatar src={userAvatar} />
        </Space>
      </div>
    </div>
  );
};

export default Header;
