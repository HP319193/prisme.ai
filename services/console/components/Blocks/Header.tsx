import { useBlock } from '@prisme.ai/design-system';
import getConfig from 'next/config';
import Link from 'next/link';
import { FC, HTMLAttributes, useCallback, useEffect, useState } from 'react';
import api from '../../utils/api';
import { EnhancedSchema } from '../SchemaForm/useSchema';

const {
  publicRuntimeConfig: { PAGES_HOST = '' },
} = getConfig();

interface Config {
  title?: string;
  logo?: {
    src: string;
    alt: string;
  };
  nav: {
    text: string;
    type: 'external' | 'internal' | 'inside' | 'event';
    value: string;
  }[];
}

const schema: EnhancedSchema = {
  type: 'object',
  properties: {
    title: {
      type: 'string',
      title: 'pages.blocks.header.settings.title.label',
      description: 'pages.blocks.header.settings.title.description',
    },
    logo: {
      type: 'object',
      title: 'pages.blocks.header.settings.logo.label',
      description: 'pages.blocks.header.settings.logo.description',
      properties: {
        src: {
          type: 'string',
          title: 'pages.blocks.header.settings.logo.src.label',
          description: 'pages.blocks.header.settings.logo.src.description',
          'ui:widget': 'upload',
        },
        alt: {
          type: 'string',
          title: 'pages.blocks.header.settings.logo.alt.label',
          description: 'pages.blocks.header.settings.logo.alt.description',
        },
      },
    },
    nav: {
      type: 'array',
      title: 'pages.blocks.header.settings.nav.label',
      description: 'pages.blocks.header.settings.nav.description',
      items: {
        type: 'object',
        title: 'pages.blocks.header.settings.nav.items.label',
        description: 'pages.blocks.header.settings.nav.items.description',
        oneOf: [
          {
            title: 'pages.blocks.header.settings.nav.items.external.label',
            description: 'pages.blocks.header.settings.nav.items.description',
            properties: {
              text: {
                type: 'string',
                title: 'pages.blocks.header.settings.nav.items.text.label',
                description:
                  'pages.blocks.header.settings.nav.items.text.description',
              },
              value: {
                type: 'string',
                title:
                  'pages.blocks.header.settings.nav.items.external.value.label',
                description:
                  'pages.blocks.header.settings.nav.items.external.value.description',
              },
            },
          },
          {
            title: 'pages.blocks.header.settings.nav.items.internal.label',
            description:
              'pages.blocks.header.settings.nav.items.internal.description',
            properties: {
              text: {
                type: 'string',
                title: 'pages.blocks.header.settings.nav.items.text.label',
                description:
                  'pages.blocks.header.settings.nav.items.text.description',
              },
              value: {
                type: 'string',
                title:
                  'pages.blocks.header.settings.nav.items.internal.value.label',
                description:
                  'pages.blocks.header.settings.nav.items.internal.value.description',
                'ui:widget': 'select:pages',
              },
            },
          },
          {
            title: 'pages.blocks.header.settings.nav.items.inside.label',
            description:
              'pages.blocks.header.settings.nav.items.inside.description',
            properties: {
              text: {
                type: 'string',
                title: 'pages.blocks.header.settings.nav.items.text.label',
                description:
                  'pages.blocks.header.settings.nav.items.text.description',
              },
              value: {
                type: 'string',
                title:
                  'pages.blocks.header.settings.nav.items.inside.value.label',
                description:
                  'pages.blocks.header.settings.nav.items.inside.value.description',
                'ui:widget': 'select:pageSections',
              },
            },
          },
          {
            title: 'pages.blocks.header.settings.nav.items.event.label',
            description:
              'pages.blocks.header.settings.nav.items.event.description',
            properties: {
              text: {
                type: 'string',
                title: 'pages.blocks.header.settings.nav.items.text.label',
                description:
                  'pages.blocks.header.settings.nav.items.text.description',
              },
              value: {
                type: 'string',
                title:
                  'pages.blocks.header.settings.nav.items.event.value.label',
                description:
                  'pages.blocks.header.settings.nav.items.event.value.description',
              },
            },
          },
        ],
        'ui:options': {
          oneOf: {
            options: [
              {
                label: 'pages.blocks.header.settings.nav.items.external.label',
                index: 0,
                value: {
                  type: 'external',
                },
              },
              {
                label: 'pages.blocks.header.settings.nav.items.internal.label',
                index: 1,
                value: {
                  type: 'internal',
                },
              },
              {
                label: 'pages.blocks.header.settings.nav.items.inside.label',
                index: 2,
                value: {
                  type: 'inside',
                },
              },
              {
                label: 'pages.blocks.header.settings.nav.items.event.label',
                index: 3,
                value: {
                  type: 'event',
                },
              },
            ],
          },
        },
      },
      'ui:options': {
        array: 'row',
      },
    },
  },
};

const LinkOrNot: FC<{ href?: string }> = ({ href, children }) => {
  if (href) {
    return (
      <Link href={href}>
        <a className="block-header__nav-item-link">{children}</a>
      </Link>
    );
  }
  return <>{children}</>;
};

const PageLink: FC<{ pageId: string } & HTMLAttributes<HTMLAnchorElement>> = ({
  pageId,
  ...props
}) => {
  const [href, setHref] = useState('');
  const fetchHref = useCallback(async (pageId: string) => {
    try {
      const { slug = pageId } = await api.getPageBySlug(pageId);
      setHref(slug);
    } catch {}
  }, []);
  useEffect(() => {
    fetchHref(pageId);
  }, [fetchHref, pageId]);
  return (
    <Link href={href}>
      <a {...props} />
    </Link>
  );
};

const Button = ({ text, type, value }: Config['nav'][number]) => {
  const { events } = useBlock<Config>();
  switch (type) {
    case 'event':
      return (
        <button
          onClick={() => {
            if (!events || !value) return;
            events.emit(value);
          }}
          className="block-header__nav-item-button"
        >
          {text}
        </button>
      );
    case 'external':
      return (
        <a href={value} className="block-header__nav-item-link">
          <button className="block-header__nav-item-button">{text}</button>
        </a>
      );
    case 'internal':
      return (
        <PageLink pageId={value} className="block-header__nav-item-link">
          <button className="block-header__nav-item-button">{text}</button>
        </PageLink>
      );
    case 'inside':
      return (
        <a href={`#${value}`} className="block-header__nav-item-link">
          <button className="block-header__nav-item-button">{text}</button>
        </a>
      );
    default:
      return null;
  }
};

export const Header = ({ edit }: { edit?: boolean }) => {
  const { config = {} as Config } = useBlock<Config>();

  const nav = config.nav && Array.isArray(config.nav) ? config.nav : [];

  return (
    <div className="block-header block-header__container flex flex-1 flex-col sm:!flex-row justify-between sm:items-center">
      <div className="block-header__left flex sm:justify-center">
        <div className="block-header__logo flex justify-center m-2 ml-4">
          {config.logo && config.logo.src && (
            // eslint-disable-next-line @next/next/no-img-element
            <img
              src={config.logo.src}
              alt={config.logo.alt}
              className="block-header__logo-image max-h-12"
            />
          )}
        </div>
        <h1 className="block-header__title flex items-center m-0 font-bold">
          {config.title}
        </h1>
      </div>
      <nav className="block-header__right flex m-4">
        {nav.map((props, k) => (
          <div
            key={k}
            className="mx-2"
            onClick={(e) => {
              if (edit) e.preventDefault();
            }}
          >
            <Button {...props} />
          </div>
        ))}
      </nav>
    </div>
  );
};

Header.schema = schema;

export default Header;
