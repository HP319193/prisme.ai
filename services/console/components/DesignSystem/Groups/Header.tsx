import { Button, Space, Avatar, Divider, Dropdown, Menu } from "../index";
import { useMemo } from "react";
import { ShareAltOutlined } from "@ant-design/icons";
import iconPrisme from "../../../icons/icon-prisme.svg";
import Image from "next/image";

interface HeaderProps {
  workspaces: string[];
  shareText: string;
  userName: string;
  userAvatar: string;
}

const Header = ({
  workspaces,
  shareText,
  userName,
  userAvatar,
}: HeaderProps) => {
  const workspacesMenu = useMemo(
    () => <Menu items={workspaces} onClick={() => {}} />,
    [workspaces]
  );

  return (
    <div className="px-6 flex flex-row w-full justify-between items-center pr-header">
      <Image src={iconPrisme} width={23} height={25} alt="prisme.ai logo" />
      <Dropdown Menu={workspacesMenu}>{workspaces[0]}</Dropdown>
      <div className="flex flex-row items-center">
        <Button type="grey">
          <Space>
            {shareText}
            <ShareAltOutlined />
          </Space>
        </Button>
        <Divider type="vertical" className="mr-4" />
        <Space>
          {userName}
          <Avatar src={userAvatar} />
        </Space>
      </div>
    </div>
  );
};

export default Header;
