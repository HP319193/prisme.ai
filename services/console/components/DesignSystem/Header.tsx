import { Button, Space, Avatar, Divider } from "./";

const Header = () => (
  <div className="px-2 flex flex-row w-full justify-between items-center pr-header border-b">
    <div>logo</div>
    <div>Nom de mon Workspace</div>
    <div className="flex flex-row items-center">
      <Button type="grey">
        <Space>
          Partager
          <i className="pi pi-share-alt" />
        </Space>
      </Button>
      <Divider type="vertical" className="mr-4" />
      <Space>
        John Doe
        <Avatar />
      </Space>
    </div>
  </div>
);

export default Header;
