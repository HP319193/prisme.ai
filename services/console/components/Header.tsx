import { ReactElement, useMemo } from 'react';
import { LogoutOutlined } from '@ant-design/icons';
import {
  Avatar,
  Divider,
  Dropdown,
  Menu,
  Space,
} from '@prisme.ai/design-system';
import { useUser } from './UserProvider';
import { useTranslation } from 'next-i18next';
import logo from '../icons/icon-prisme.svg';
import Image from 'next/image';
import Link from 'next/link';

interface HeaderProps {
  title?: string | ReactElement;
  leftContent?: string | ReactElement;
}
const Header = ({ title, leftContent }: HeaderProps) => {
  const { t } = useTranslation('user');
  const { user, signout } = useUser();

  const userMenu = useMemo(
    () => (
      <Menu
        items={[
          {
            label: (
              <Space>
                <LogoutOutlined />

                {t('sign.out')}
              </Space>
            ),
            key: 'signout',
          },
        ]}
        onClick={(item) => {
          if (typeof item === 'string') return;
          switch (item.key) {
            case 'signout':
              signout();
          }
        }}
      />
    ),
    [signout, t]
  );

  return (
    <div className="relative px-6 flex flex-row w-full justify-between items-center pr-header z-20 bg-prisme-darkblue text-white">
      <Link href="/workspaces">
        <a>
          <Image {...logo} width="25px" height="20px" alt="Prisme.ai" />
        </a>
      </Link>
      {title}
      <div className="flex flex-row items-center">
        {leftContent}

        {leftContent && <Divider type="vertical" className="mr-4" />}

        {user && (
          <Dropdown Menu={userMenu} placement="bottomRight" arrow={false}>
            <Space>
              {user.firstName}
              {user.photo && <Avatar src={user.photo} />}
            </Space>
          </Dropdown>
        )}
      </div>
    </div>
  );
};

export default Header;
