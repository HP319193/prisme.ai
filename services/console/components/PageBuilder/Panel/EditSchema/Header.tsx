import { Schema } from '@prisme.ai/design-system';

export const schema: Schema = {
  type: 'object',
  properties: {
    title: {
      type: 'string',
      title: 'pages.blocks.header.settings.title.label',
      description: 'pages.blocks.header.settings.title.description',
    },
    logo: {
      type: 'object',
      title: 'pages.blocks.header.settings.logo.label',
      description: 'pages.blocks.header.settings.logo.description',
      properties: {
        src: {
          type: 'string',
          title: 'pages.blocks.header.settings.logo.src.label',
          description: 'pages.blocks.header.settings.logo.src.description',
          'ui:widget': 'upload',
        },
        alt: {
          type: 'string',
          title: 'pages.blocks.header.settings.logo.alt.label',
          description: 'pages.blocks.header.settings.logo.alt.description',
        },
      },
    },
    nav: {
      type: 'array',
      title: 'pages.blocks.header.settings.nav.label',
      description: 'pages.blocks.header.settings.nav.description',
      items: {
        type: 'object',
        title: 'pages.blocks.header.settings.nav.items.label',
        description: 'pages.blocks.header.settings.nav.items.description',
        properties: {
          type: {
            hidden: true,
          },
          text: {
            type: 'string',
            title: 'pages.blocks.header.settings.nav.items.text.label',
            description:
              'pages.blocks.header.settings.nav.items.text.description',
          },
        },
        oneOf: [
          {
            title: 'pages.blocks.header.settings.nav.items.external.label',
            description: 'pages.blocks.header.settings.nav.items.description',
            properties: {
              value: {
                type: 'string',
                title:
                  'pages.blocks.header.settings.nav.items.external.value.label',
                description:
                  'pages.blocks.header.settings.nav.items.external.value.description',
              },
            },
          },
          {
            title: 'pages.blocks.header.settings.nav.items.internal.label',
            description:
              'pages.blocks.header.settings.nav.items.internal.description',
            properties: {
              value: {
                type: 'string',
                title:
                  'pages.blocks.header.settings.nav.items.internal.value.label',
                description:
                  'pages.blocks.header.settings.nav.items.internal.value.description',
                'ui:widget': 'select',
                'ui:options': {
                  from: 'pages',
                },
              },
            },
          },
          {
            title: 'pages.blocks.header.settings.nav.items.inside.label',
            description:
              'pages.blocks.header.settings.nav.items.inside.description',
            properties: {
              value: {
                type: 'string',
                title:
                  'pages.blocks.header.settings.nav.items.inside.value.label',
                description:
                  'pages.blocks.header.settings.nav.items.inside.value.description',
                'ui:widget': 'select',
                'ui:options': {
                  from: 'pageSections',
                },
              },
            },
          },
          {
            title: 'pages.blocks.header.settings.nav.items.event.label',
            description:
              'pages.blocks.header.settings.nav.items.event.description',
            properties: {
              value: {
                type: 'string',
                title:
                  'pages.blocks.header.settings.nav.items.event.value.label',
                description:
                  'pages.blocks.header.settings.nav.items.event.value.description',
              },
            },
          },
        ],
        'ui:options': {
          oneOf: {
            options: [
              {
                label: 'pages.blocks.header.settings.nav.items.external.label',
                index: 0,
                value: {
                  type: 'external',
                },
              },
              {
                label: 'pages.blocks.header.settings.nav.items.internal.label',
                index: 1,
                value: {
                  type: 'internal',
                },
              },
              {
                label: 'pages.blocks.header.settings.nav.items.inside.label',
                index: 2,
                value: {
                  type: 'inside',
                },
              },
              {
                label: 'pages.blocks.header.settings.nav.items.event.label',
                index: 3,
                value: {
                  type: 'event',
                },
              },
            ],
          },
        },
      },
      'ui:options': {
        array: 'row',
      },
    },
  },
};

export default schema;
