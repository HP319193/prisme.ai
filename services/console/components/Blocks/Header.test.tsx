import Header from './Header';
import renderer from 'react-test-renderer';

jest.mock('@prisme.ai/design-system', () => {
  const mock = {
    config: {
      title: 'Hello',
      logo: {
        src: 'http://img/age.png',
        alt: 'hello logo',
      },
      nav: [
        {
          text: 'Home',
          type: 'external',
          value: 'https://prisme.ai',
        },
        {
          text: 'Another page',
          type: 'internal',
          value: 'j8ksz9',
        },
        {
          text: 'Scroll',
          type: 'inside',
          value: 'section_1',
        },
        {
          text: 'Contact',
          type: 'event',
          value: 'displayContactForm',
        },
      ],
    },
  };
  return {
    useBlock: () => mock,
  };
});

it('should render', () => {
  const root = renderer.create(<Header />);
  expect(root.toJSON()).toMatchSnapshot();
});
