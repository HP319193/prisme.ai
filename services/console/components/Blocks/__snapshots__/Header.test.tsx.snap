// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="block-header block-header__container flex flex-1 flex-col md:!flex-row justify-between md:items-center"
>
  <div
    className="block-header__left flex md:justify-center"
  >
    <div
      className="block-header__logo flex justify-center m-2 ml-4"
    >
      <img
        alt="hello logo"
        className="block-header__logo-image max-h-12"
        src="http://img/age.png"
      />
    </div>
    <h1
      className="block-header__title flex items-center m-0 font-bold"
    >
      Hello
    </h1>
  </div>
  <nav
    className="block-header__right flex m-4"
  >
    <div
      className="mx-2"
      onClick={[Function]}
    >
      <a
        className="block-header__nav-item-link"
        href="https://prisme.ai"
      >
        <button
          className="block-header__nav-item-button"
        >
          Home
        </button>
      </a>
    </div>
    <div
      className="mx-2"
      onClick={[Function]}
    >
      <a
        className="block-header__nav-item-link"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <button
          className="block-header__nav-item-button"
        >
          Another page
        </button>
      </a>
    </div>
    <div
      className="mx-2"
      onClick={[Function]}
    >
      <a
        className="block-header__nav-item-link"
        href="#section_1"
      >
        <button
          className="block-header__nav-item-button"
        >
          Scroll
        </button>
      </a>
    </div>
    <div
      className="mx-2"
      onClick={[Function]}
    >
      <button
        className="block-header__nav-item-button"
        onClick={[Function]}
      >
        Contact
      </button>
    </div>
  </nav>
</div>
`;
